patterns/three_levels_of_thinking.md

```diff
@@ -32,53 +32,58 @@ The method signatures are the verbs of your programming literature.
 Inside the methods are the implementation.  They draw their context from the method signatures inside the context of a class.
 
 This is where the sentences and paragraphs are put together.  Messages are sent to objects.
-If _Intention Revealing Names_ are used for classes and method signatures,
-the implementation can read almost like natural language.
+
+If you use _Intention Revealing Names_ for classes and method signatures, the implementation can read almost like natural language. Really.
 
 ## Solution
 
 In order to allow readers to grok your code at the Conceptual, Specification, and Implementation levels,
 You must write your code with these three levels of thinking in mind at all times.
+* _Conceptual_ should be clear from reading class names alone.
+* _Specification_ should be clear from reading method signatures.
+* _Implementation_ should be clear from reading method details that follow the conceptual and specification naming schemes.
 
 Classes should be named to convey their purpose and answer the question, `What is this component responsible for?`
-Typically these are nouns or short noun phrases that further qualify the context in which the main noun is appropriate.
+Typically these are nouns or short noun phrases that further qualify the context the main noun is appropriate in. For example, you might have a class hierarchy like this:
+```
+Frame
+    |
+  StructuralFrame
+        |        \
+    RoofFrame    DeckFrame
+```
+In this example, StructuralFrame's context and singular responsibility should be a subset of Frame's context and responsibility. Likewise, DeckFrame's and RoofFrame's context and responsibility should be a subset of StructuralFrame's.
 
 Method signatures should indicate `What command does this execute?` or `What question does this answer?` without having to read the implementation details.
-Parameters should have type suggesting names.
+Parameter's names should be suggestive of their type (e.g. ```name``` suggests a string, not an array).
 
-The internal code of a method should exploit the power of the conceptual class names and the specification level method signatures
-with role suggesting variable names.
-Instance variables (or properties) and the implementation of methods answer the question `What information does the component manage?` and
-`How does it execute the commands and answer the questions specified in its interface?`
+Once you are working on the implementation of a method, exploit the power of class names (conceptual thinking level) and method signatures (specification thinking level). You can exploit the power of these two things by using instance and local variable names suggestive of their role (implementation thinking level).
```

Comment:
Agree there should be some rewording here, but this isn't it.  Let's work on this together.